src/lib/providers/TransferProvider.tsx

```diff
@@ -0,0 +1,77 @@
+//- React Imports
+import React, { useEffect, useState } from 'react';
+
+//- Type Imports
+import useNotification from 'lib/hooks/useNotification';
+
+export interface TransferSubmitParams {
+	name: string;
+	domain: string; // domain label
+	image: string;
+	creatorId: string;
+	walletAddress: string;
+}
+
+export const TransferContext = React.createContext({
+	transferring: [{}],
+	transferred: [{}],
+	submit: (params: TransferSubmitParams) => {},
+});
+
+type TransferProviderType = {
+	children: React.ReactNode;
+};
+
+const TransferProvider: React.FC<TransferProviderType> = ({ children }) => {
+	const { addNotification } = useNotification();
+	const [transferring, setTransferring] = useState<TransferSubmitParams[]>([]);
+	const [transferred, setTransferred] = useState<TransferSubmitParams[]>([]);
+	const [
+		finishedTransferring,
+		setFinishedTransferring,
+	] = useState<TransferSubmitParams | null>(null);
+
+	const submit = async (params: TransferSubmitParams) => {
+		console.log(params);
+		addNotification(`Started transfer`);
+		setTransferring([...transferring, params]);
+
+		const finishTransferring = async () => {
+			setFinishedTransferring(params);
+		};
+
+		// TODO
```

Comment:
Make sure to indicate what TODO's are please, or what it should be doing instead

Reply:
@Remscar I've cleaned up this. See latest commit